### DISCUSSION TITLE
WPF: Create Tray without Window

### DISCUSSION THREAD

**Discussion by PuerNoctis** [HavenDV/H.NotifyIcon#56] [Q&A] [answered]
Hi!

As per title. How do I create a tray icon with a WPF application, without having to show a window on startup? When I add the tray icon into the main window, it is only shown when the Window's Visibility-Property is actually marked 'Visible'. If I set it to 'Hidden' the tray doesn't show.

I want to start a tray application with the initial Window being hidden.

With traditional WinForms, trays seemed less interlinked with the Window itself, and it worked like charm.

Thanks for any suggestions!

**Comment by HavenDV** [comment, marked as answer]
Please check this code: https://github.com/HavenDV/H.NotifyIcon/tree/master/src%2Fapps%2FH.NotifyIcon.Apps.Wpf.Windowless

**Reply by PuerNoctis** [reply]
Thank you, that's perfect!
